src/controllers/hid/hiddevice.h

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <hidapi.h>
+
+#include <QObject>
+#include <string>
+
+class ProductInfo;
+
+namespace mixxx {
+
+namespace hid {
+
+constexpr unsigned short kAppleVendorId = 0x5ac;
+
+class DeviceInfo final {
+  public:
+    explicit DeviceInfo(
+            const hid_device_info& device_info);
```

Comment:
The similar naming of this and the hid_device_info class from the hidapi library is confusing. What is the difference between them? What is the need for our own wrapper around hidapi's class? Please add a comment above the class declaration explaining these.

Reply:
IMHO same things should be refered by same terms. This is just a C++ wrapper around a C API.
The USB "Device Class Definition for HID" specification refers these as:
a.) Device Descriptor (VendorID & ProductID)
b.) Interface Descriptor (Interface Number)

Instead of inventing a new name, we should either:
-use the HIDAPI terms
-use the USB class compliant terms

Reply:
If you are aware of the official naming please comment directly the affected parts of the code and provide a proposed change. This not only applies not only to the naming but also to the formatting of the values. Links to the respective sources are appreciated, then we can add them as comments.

Reply:
The USB specifications are public: https://www.usb.org/document-library/device-class-definition-hid-111
I'm not an USB expert, but I guess, that Appendix E1 and E3 describe these items.

Reply:
I'm fine with keeping the HIDAPI term DeviceInfo - but I'm against to invent an own term.

Reply:
I know the specs and only skimmed them quickly to verify or fix some terms. No time to read them in detail.

Reply:
The original name I used actually was `DeviceDescriptor`. But since it resembles `hid_device_info` I switched back to `DeviceInfo` for consistent naming.

Reply:
Maybe important for the future controller system is, that an USB device has always one Device Descriptor, but can have multiple interfaces, like Audio, HIDAPI or Bulk for embedded displays.

Reply:
I'm not objecting to the name. I'm just asking for a little explanation of the purpose of the class.